src/rpdk/core/contract/resource_client.py

```diff
@@ -334,6 +334,29 @@ def generate_invalid_update_example(self, create_model):
         example = override_properties(self.invalid_strategy.example(), overrides)
         return {**create_model, **example}
 
+    def compare(self, inputs, outputs):
+        assertion_error_message = (
+            "All properties specified in the request MUST "
+            "be present in the model returned, and they MUST"
+            " match exactly, with the exception of properties"
+            " defined as writeOnlyProperties in the resource schema"
+        )
+        try:
+            for key in inputs:
+                if isinstance(inputs[key], dict):
+                    self.compare(inputs[key], outputs[key])
+                elif isinstance(inputs[key], list):
```

Comment:
is `list` safe to go with (no sets)? should you change it to `Collection`?

Reply:
in my opinion it should be list. As we are dealing with an order list of items ```[]``` vs set being an unordered list ```{}``` 
Is there a specific example where you might see this to be probromatic?

Reply:
I think since it's json it's always either dict or list